src/Microsoft.Identity.Web/TokenAcquisition.cs

```diff
@@ -48,12 +50,14 @@ internal class TokenAcquisition : ITokenAcquisition, ITokenAcquisitionInternal
             IHttpContextAccessor httpContextAccessor,
             IOptions<MicrosoftIdentityOptions> microsoftIdentityOptions,
             IOptions<ConfidentialClientApplicationOptions> applicationOptions,
+            IHttpClientFactory httpClientFactory,
```

Comment:
Can this ever be null? I don't understand the DI that well, so apologies for asking silly questions.

Reply:
No it cannot, @bgavrilMS:  The [ASP] .NET Core framework will throw if an `IHttpClientFactory` was not present as a service. It's presence is ensured in Microsoft.Identity.Web by the call to `services.AddHttpClient()`which was added in the `WebApiAuthenticationBuilderExtensions` and `WebAppAuthenticationBuilderExtensions extension methods`

Reply:
The framework resolves these references recursively. So if a service DI mapping was not added for some deep dependency, it will throw an error.